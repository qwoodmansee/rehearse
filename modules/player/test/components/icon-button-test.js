import BackToPoint from '@player/assets/back-to-point.png';
import IconButton from '@player/src/components/icon-button';
import React from 'react';
import RehearseText from '@core/src/components/rehearse-text';
import { fireEvent, render } from '@testing-library/react-native';

describe('IconButton', () => {
  it('renders the icon', () => {
    const { baseElement } = render(
      <IconButton
        icon={BackToPoint}
        onPress={() => {}}
      />);

    expect(baseElement).toMatchSnapshot();
  });

  it('calls the onPress function when pressed', () => {
    const mockPressFn = jest.fn();
    const { getByTestId } = render(
      <IconButton
        icon={BackToPoint}
        onPress={mockPressFn}
        testID={'test-control'}
      />);
    fireEvent.press(getByTestId('test-control'));
    expect(mockPressFn).toHaveBeenCalled();
  });

  describe('when passed text', () => {
    it('renders the text', () => {
      const { getByText } = render(<IconButton><RehearseText>ButtonText</RehearseText></IconButton>);
      getByText('ButtonText');
    });
  });
});
