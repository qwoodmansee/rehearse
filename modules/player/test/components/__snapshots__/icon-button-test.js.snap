// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders the icon 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#f05545",
            "borderRadius": 20,
            "height": 80,
            "justifyContent": "center",
            "padding": 10,
            "position": "relative",
            "width": 80,
          },
          Object {},
        ]
      }
      testID="internal-button"
    >
      <Image
        source={
          Object {
            "testUri": "../../../modules/player/assets/back-to-point.png",
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;
